pkgs/development/python-modules/argon2-cffi/default.nix

```
@@ -1,42 +1,36 @@
-{ cffi
-, six
-, enum34
-, hypothesis
+{ hypothesis
 , pytest
-, wheel
 , buildPythonPackage
 , fetchPypi
-, isPy3k
 , lib
-, stdenv
+, hatchling
+, hatch-vcs
+, hatch-fancy-pypi-readme
 , argon2-cffi-bindings
 }:
 
 buildPythonPackage rec {
   pname = "argon2-cffi";
-  version = "21.3.0";
-  format = "flit";
+  version = "23.1.0";
+  format = "pyproject";
 
   src = fetchPypi {
-    inherit pname version;
-    sha256 = "d384164d944190a7dd7ef22c6aa3ff197da12962bd04b17f64d4e93d934dba5b";
+    pname = "argon2_cffi";
+    inherit version;
+    hash = "sha256-h5w+eaJynOdo67fTbUYJ46eKTKLsOp8SKGygV+PQ2wg=";
   };
 
-  propagatedBuildInputs = [ cffi six argon2-cffi-bindings ]
-    ++ lib.optional (!isPy3k) enum34;
+  nativeBuildInputs = [ hatchling hatch-vcs hatch-fancy-pypi-readme ];
 
-  propagatedNativeBuildInputs = [
-    argon2-cffi-bindings
-    cffi
-  ];
+  propagatedBuildInputs = [ argon2-cffi-bindings ];
 
-  ARGON2_CFFI_USE_SSE2 = lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) "0";
-
-  nativeCheckInputs = [ hypothesis pytest wheel ];
+  nativeCheckInputs = [ hypothesis pytest ];
```

```suggestion
  nativeCheckInputs = [ hypothesis pytestCheckHook ];
```
Would you please change pytest to pytestCheckHook?